xdsl/parser/attribute_parser.py

```diff
@@ -718,13 +813,21 @@ def _parse_builtin_dense_attr(self, _name: Span) -> DenseIntOrFPElementsAttr:
             self.raise_error("Dense literal attribute should have a static shape.")
 
         element_type = type.element_type
+        data_values: list[int] | list[float] = []
         # Convert list of elements to a list of values.
-        if shape != []:
-            data_values = [value.to_type(self, element_type) for value in values]
+        # For hex string, the data values are already constructed
```

Comment:
Is this true? Isn't one of your examples a `2xi32` array where there's a single repeated value decoded?